### PR TITLE
Homework #2 Variant #11

### PR DESCRIPTION
**Pull request by Mifistoff**

### CODE REVIEW

File: lib/topics/state_and_behaviour.rb [hunk 1]
```diff
@@ -2,4 +2,31 @@
...
+    end
+
+    # create new Car object with default parameters
+    def self.default_car
+      new
+    end
+
+    # returns decremented current speed if this possible
+    def push_break(speed_down_value)
+      return unless @current_speed >= speed_down_value
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: You have attr reader, don't you?

Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: Use `if` here. It is easier to read.

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -2,4 +2,31 @@
...
+    # initialize for Car object
+    def initialize(car_praremeters)
+      @year = car_praremeters[:year] || 2015
+      @color = car_praremeters[:color] || 'purple'
+      @model = car_praremeters[:model] || 'skyline'
+      @current_speed = 0
+    end
+
+    # create new Car object with default parameters
+    def self.default_car
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: According to style guide, class methods have to be placed before instance methods

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -4,23 +4,23 @@
...
-    end
-
+    attr_accessor :color, :model, :year, :current_speed
     # create new Car object with default parameters
     def self.default_car
       new
     end
 
+    # initialize for Car object
+    def initialize(car_parameters)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: car_params = {}

File: lib/topics/state_and_behaviour.rb [hunk 4]
```diff
@@ -4,23 +4,23 @@
...
+      @year = car_parameters[:year] || 2015
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: try .fetch and default values move to CONSTANTS

File: lib/topics/state_and_behaviour.rb (hunk 3 above)
Comment (Mifistoff) [reply lib/topics/state_and_behaviour.rb]: 1) Failure:
Car::#initialize#test_0001_expects hash of attributes [/var/www/pivorak/homework2/test/topics/test_state_and_behaviour.rb:26]:
ArgumentError expected but nothing was raised.

15 runs, 18 assertions, 1 failures, 0 errors, 0 skips

Comment (Mifistoff) [reply lib/topics/state_and_behaviour.rb]: Test fails on this change.

**Pull request closed without merging**